### ISSUE TITLE
Feature request for selecting NVDA features to copy to a portable NVDA copy

### ISSUE THREAD

**Issue by tarikhadzirovicofficial** [nvaccess/nvda#16992]
### Describe the solution you'd like
NVDA has a great feature that other screen readers don't have, which allows it to be used on computers that don't have any screen reader installed. We know that running a backup is not possible from read-only media, but that's not the topic right now; after all, media like CDs are used less and less today.
In the dialog box for creating a portable copy, there are the following options:
* No label (edit field),
* Browse (button),
* Copy current user configuration (checkbox),
* Start the new portable copy after creation (checkbox).

What I noticed while writing this article is that the edit field does not have a label, but that's not very important. However, if the NVDA developers wouldn't mind, they could add one. But that's not what I want to discuss.
It would be nice if users could choose which parts of NVDA will be copied. If the checkbox for copying configuration settings is checked, a list of NVDA features that can be copied should appear, such as settings, add-ons, configuration profiles, input gestures, speech dictionaries, punctuation/symbol pronunciation, etc. This list should look similar to those during the installation of Notepad++ or PotPlayer, where we can choose a full, minimal, or custom installation. When custom is selected, we can choose what we want to install.
It would also be nice if we could select which categories of NVDA settings will be copied. For example, speech, Braille, vision, mouse...
I think it's easier to determine what we want to transfer to the portable NVDA copy this way, so we don't have to configure it and remove things we don't need.

**Comment by XLTechie** [comment]
@tarikhadzirovicofficial, I find this idea intriguing.

I wonder if an "Advanced" button might be a good idea, so as not to clutter the dialog for people who just want to check the one box and copy everything?

Doing this might also lead to a general framework for solving #6305.

**Comment by tarikhadzirovicofficial** [comment]
@XLTechie I agree that an "Advanced" or "Additional Options" button is a better idea.

**Comment by seanbudd** [comment]
Closing this as won't fix

This issue missed an extremely important part of the template, describe the issue you are having. It is unclear the use case for this.
If you have a config set up, why would you want a separate config for a portable copy?

**Comment by tarikhadzirovicofficial** [comment]
Hi @seanbudd!
I omitted that part, thinking that the need for this feature was clear. Why shouldn't a user be able to select which parts of the program to copy? For example, consider only the add-ons. The user does not want to include add-ons, but if they uncheck the option to copy the entire configuration, nothing will be copied. They still want to copy the settings but not the add-ons. The user has 30+ add-ons. Removing each one individually would be tedious, and going to the NVDA installation folder for regular users is not the best solution. I believe that the ability to selectively choose which program features the user wants in the portable copy is crucial for efficient operation, especially for users who frequently handle such tasks. Since I often service computers, I frequently create portable copies. However, I do not want configuration profiles and punctuation and symbol pronunciation to be copied. I also do not need all the add-ons, so a feature to select which add-ons to copy would be useful, but I won’t go into that now.

**Comment by seanbudd** [comment]
The case for add-ons is covered in #2467 and #9020.

Please provide some reasoning for being able to specify specific parts of configuration when copying.
It's not immediately clear why you would only want to copy a part of your config. If you are a user of NVDA, and happy with your default config, why would you want a different portable copy with a different configuration?

Note that you mention  copying "program features" or "parts of the program", however you are describing copying configuration. When you create a portable copy, we must include the entire copy of NVDA, it's not subdividable into component features like NotePad++ or the other cases your describe with program installers. Of course, we can filter out add-ons, but outside of this, the case you're expecting isn't clear. We're not going to create a case where you can create a portable copy of NVDA without features like input gestures or the add-on store.



See also #6969

**Comment by tarikhadzirovicofficial** [comment]
Hi @seanbudd!
I understand that I may have given a silly example when I mentioned installing the Notepad++ program. However, I believe I still provided a good example in the context of displaying that menu. 
I've already explained why I would like this feature, but I'll try to be more detailed here. 
I'm satisfied with my NVDA configuration, and there are no issues there. When I want to change something, I'll change it. However, I also work on servicing computers, and I want my portable copies of NVDA to be up-to-date and include all the settings I might change over time. Since there's no way to automatically create a backup at a specific time, I have to do this manually about once a month. Specifically, I use 54 add-ons, and I don't need 80% of them on that portable copy. When I create a portable copy, I have to deal with deleting all the add-ons I don't need (if I've checked the option to copy the configuration), or I have to set up NVDA and install the necessary add-ons, then configure them, and so on (if I haven't checked the option to copy the configuration). I understand that I can copy the nvda.ini file, but I'm afraid that's not a permanent solution. If I have to spend half an hour editing the portable copy each time, then what's the point of all that work if I have to repeat it in a month? Besides, I think that for other users who aren't as skilled (I was once like that myself), copying the nvda.ini file could be complicated.
Regarding other settings: I think transparency is very important here. Let's exclude add-ons and the selection of which add-on will be copied and which won't. For example, why would I need speech dictionaries for Unigram, WhatsApp, and a bunch of other applications when I won't be using them on my portable copy for servicing computers? Why would I need a bunch of input gestures that I don't use? But okay, I can somewhat understand the lack of need for the option to choose whether to copy input gestures or not, since if I'm servicing a computer and using my own shortcuts, I want to have them there for easier work, instead of having to remember both the default NVDA shortcuts and my customized ones.
Why would I need profiles for Brave, Playroom, Thunderbird, Word, and other applications? I have many more examples, but I'll stop here for now.

**Comment by seanbudd** [comment]
@tarikhadzirovicofficial - copying that additional configuration doesn't seem to be detrimental to your experience? it seems to have no effect at all except taking up a few kB worth of space at most. And you have the bonus of having that config set up if you ever need it. A much more complicated user experience doesn't seem worthwhile to save a few hundred lines of text for a 40MB application

**Comment by tarikhadzirovicofficial** [comment]
Hi @seanbudd!
Thanks for your response, but I feel like I've already explained that those additional things, like user profiles, are not necessary for, for example, my portable copies. It seems like there's a misunderstanding here. If a user like myself has the option to create a portable copy, it makes sense to want to customize it right from the start, deciding what is needed and what isn't. No one would delete configuration profiles from their main configuration just to create a portable copy, nor would they want to go through the hassle of doing so after the portable copy is made.
I emphasize again that for basic users, the process of removing such things is complex, and I believe implementing this feature wouldn't be too demanding. It would allow users to have more control and a streamlined experience when creating portable copies.
Thanks for considering this.
